### ISSUE TITLE
Runtime failure in H5T__init_native_float_types() 

### ISSUE THREAD

**Issue by dkokron** [HDFGroup/hdf5#4381]
**Describe the bug**
I'm seeing a floating point exception while opening a NetCDF4 file with nf90_open().  A partial stack trace follows.

```
 forrtl: error (65): floating invalid
 Image              PC                Routine            Line        Source
 gsi.x              0000000008BF578B  Unknown               Unknown  Unknown
 libpthread-2.31.s  000014D77073D8C0  Unknown               Unknown  Unknown
 gsi.x              0000000008721303  H5T__init_native_         531  H5Tinit_float.c
 gsi.x              0000000008559851  H5T_init                  751  H5T.c
 gsi.x              0000000008796EC0  H5VL_init_phase2          198  H5VLint.c
 gsi.x              0000000007E49B20  H5_init_library           268  H5.c
 gsi.x              000000000804A0F5  H5Eset_auto2             1508  H5E.c
 gsi.x              0000000007DBEE3D  set_auto                   64  hdf5internal.c
 gsi.x              0000000007DBEE62  nc4_hdf5_initiali          77  hdf5internal.c
 gsi.x              0000000007DBC0DB  NC_HDF5_initializ         128  hdf5dispatch.c
 gsi.x              0000000007E0C4B5  nc_initialize             107  nc_initialize.c
 gsi.x              0000000007D5BC4B  NC_open                  1983  dfile.c
 gsi.x              0000000007D5ABFA  nc_open                   668  dfile.c
 gsi.x              0000000007C2EAF6  nf_open_                  189  nf_control.F90
 gsi.x              0000000007C4FA8C  netcdf_mp_nf90_op          64  netcdf4_file.F90
 gsi.x              0000000000EA6CEC  gsi_rfv3io_mod_mp         339  gsi_rfv3io_mod.f90
```

**Expected behavior**
otta work

**Platform (please complete the following information)**
 - HDF5 version
1.14.3
 - OS and version
SLES-15sp3
 - Compiler and version
ftn and cc from the Cray build environment.
which cc
/opt/cray/pe/craype/2.7.17/bin/cc
cc --version
icc (ICC) 19.1.3.304 20200925
which ftn
/opt/cray/pe/craype/2.7.17/bin/ftn
ftn --version
ifort (IFORT) 19.1.3.304 20200925
 - Build system (e.g. CMake, Autotools) and version
 Autotools
 - Any configure options you specified
export FFLAGS="-g -O0 -traceback -fPIC -w"
export CFLAGS="-g -O0 -traceback -fPIC -w"
export CXXFLAGS="-fPIC -w"
export FCFLAGS="$FFLAGS"

--enable-fortran --enable-cxx --enable-build-mode=debug --with-zlib=/lfs/h1/hpc/support/daniel.kokron/Projects/hpc-stack/install/intel-19.1.3.304/zlib/1.2.11 --disable-shared --enable-static --enable-parallel --enable-unsupported

 - MPI library and version (parallel HDF5)
 cray-mpich-8.1.20

**Additional context**
[config.log](https://github.com/HDFGroup/hdf5/files/14937325/config.log)

**Comment by Krande** [comment]
Hey, 

I believe this might be related to https://github.com/HDFGroup/hdf5/issues/3831?

FYI: We experienced the same in https://github.com/conda-forge/code-aster-feedstock/pull/57. I re-ran the compilation with hdf5=1.14.0 locally and the issue was gone. 

Hopefully this will be fixed in [1.14.4](https://github.com/HDFGroup/hdf5/milestone/5) (which I believe has a milestone freeze today).

**Comment by derobins** [comment]
This should be fixed in 1.14.4, which releases today. The floating point exceptions are generated by our datatype introspection code, which runs at library startup. This used to be done in a separate program at compile time, but this made cross-compilation difficult, so now it's a runtime check.

Please reopen if this isn't fixed in 1.14.4 and we'll investigate further

**Comment by dkokron** [comment]
Just tested with the hdf5_1_14_4 branch.  The "floating invalid" error is gone.
